neusomatic/python/utils.py

```diff
@@ -13,28 +13,26 @@
 
 
 FORMAT = '%(levelname)s %(asctime)-15s %(name)-20s %(message)s'
-logFormatter = logging.Formatter(FORMAT)
+logging.basicConfig(level=logging.INFO, format=FORMAT)
 logger = logging.getLogger(__name__)
-consoleHandler = logging.StreamHandler()
-consoleHandler.setFormatter(logFormatter)
-logger.addHandler(consoleHandler)
-logging.getLogger().setLevel(logging.INFO)
 
 
-def run_shell_command(command, stdout=None, stderr=None):
+def run_shell_command(command, stdout=None, stderr=None, run_logger=None):
     stdout_fd = open(stdout, "w") if stdout else None
     stderr_fd = open(stderr, "w") if stderr else None
+    my_logger = logger
```

Comment:
Can't you just have `my_logger = logging.getLogger(...)` where the name is the function name, instead of passing the logger?

Reply:
we need to pass logger here so that we know which script has called run_shell_command